paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -941,6 +1014,102 @@ void DistSSAGraphBuilder::InsertPostprocessOps(ir::Graph *result) const {
   }
 }
 
+void FuseAllReduceSSAGraphBuilder::CheckGraph(const ir::Graph &graph) const {
```

Comment:
Why need to add a new `FuseAllReduceSSAGraphBuilder`? maybe put fuse passes after multi device graph pass can reduce number of builders here?